src/main/java/me/ryanhamshire/GPFlags/util/MessagingUtil.java

```diff
@@ -52,7 +51,7 @@ public static void sendMessage(@Nullable CommandSender receiver, String message)
             message = PlaceholderApiHook.addPlaceholders(player, message);
         } catch (Throwable ignored) {}
         message = message.replace(COLOR_CHAR, '&');
-        Component component = MiniMessage.miniMessage().deserialize(message);
+        Component component = ChatUtil.validatedHexComp(receiver, message);
```

Comment:
I agree. this was my initial idea but as you already mentioned: It could become a bit difficult.
However, I think Ill figure it out and replace it so only the setters stuff will be directly changed on saving

Reply:
I figured out that we can change the methods return value from Component into String and check for those validations in the setFlag method. How does that sound?